src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -364,6 +364,11 @@ def _monitor_workers_and_process_tasks_in_thread(
         # If the while loop exits due to batch run timeout, we should set is_timeout to True if we didn't set it before.
         self._is_timeout = self._is_timeout or self._batch_timeout_expired(batch_start_time)
 
+        # wait 10 seconds to ensure the spans are exported before the process is killed
+        # otherwise there will be some missing spans for lines
+        # TODO (Task 2950080): make line process wait for spans flush
+        time.sleep(10)
```

Comment:
MARK & highlight: this change might lead to check-enforcer timeout/CI break, as we make every line 10 seconds longer

Reply:
added a check before the sleep, however, I believe all batch run from SDK should internally set up the exporter, so the check shall not work better.